src/ui/MainWindow.cc

```diff
@@ -274,6 +274,11 @@ MainWindow::MainWindow()
 
 MainWindow::~MainWindow()
 {
+    // Enforce thread-safe shutdown of the mavlink decoder
+    mavlinkDecoder->finish();
+    mavlinkDecoder->wait(1000);

```

Comment:
A bit concerned about slowing down shutdown even more. I think you can do a wait using MultiSignalSpy on the thread closing out can't yet. I think there is some sort of signal there you can wait on?